docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+
+### Templating
+
+Chosen option: "Option B - Use React server-side to generate static pages", because:
+
+- The team has experience with (and likes) React
+- We're already using it in other parts of the FxA ecosystem
+- React has been tried and tested to work well with Storybook
+- Components across the templates can be reused if written in React
+
+A note about tests: Since we're going with the React-based solution for our templating stack, it may make sense to use Jest to test the templates. However, we don't want a ton of time spent in test land. At the very least, we can take what we currently have and transfer/copy it, and look them over to see if anything can be improved without a high level of effort.
+
+#### MJML
+
+We have decided to not use MJML as part our templating stack, because:
+
+- MJML's claim is to code responsive emails easily but FxA emails are currently not designed to need much in terms of responsiveness and it'd add a layer of complication we wouldn't benefit much from during a mustache to React conversion.
+
+### CSS Styles
+
+Chosen option: Use TailwindCSS, because:
+
+- It is being used in the FxA ecosystem so the engineers have prior experience with it and also like it.
+- There is also a style guide written for `fxa-settings` which leads us to a possibility of reusing it for our emails.
+- Tailwind classes can be converted into inline styles with the help of a library like [juice](https://github.com/Automattic/juice)
+- TailwindCSS provides us the opportunity to work with single-purpose classes as well as component classes which can help us achieve reusability and also DRY up our codebase.
+- Tailwind is highly configurable without being cumbersome, allowing us to modify type and spacing scales, define color ranges, and set up media queries to meet our exact needs.
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and provides us the flexibility to preview all of the different email states together.
+- We have a Storybook deployment already in place and has worked great so far for a few other FxA services so hooking up the client for `fxa-auth-server` to view the modernized components might result in consistency.
+- Easy to test and perform QA. The design and product teams don't have to touch the codebase to test various states of UI.
+- Support for CSS (whether it's plain old CSS, TailwindCSS or something else)
+
+Note: Along with using storybook to view the various states of the templates, we are also planning on continued support of [Maildev](https://www.npmjs.com/package/maildev) since it has applications beyond just previewing emails.
+
+### Other
+
+**Plaintext files**: We'll be converting our React templates over to HTML and once the HTML templates are generated we can extract the text out of them using a HTML parser like [html-to-text](https://github.com/html-to-text/node-html-to-text) or [textversionjs](https://github.com/EDMdesigner/textversionjs)
```

Comment:
Interesting choice.  I wonder how well this is going to work with some header/footer content.  It does indirectly reduce l10n effort since the extracted text content will be localized.

Reply:
I personally haven't vetted either of these options, and wondered if what we go with here might change once we get into it anyway, so thank you for your thoughts and notes on gotchas. 😆